sdk/communication/Azure.Communication.CallAutomation/src/Models/StartDialogOptions.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+
+namespace Azure.Communication.CallAutomation
+{
+    /// <summary>
+    /// The Dialog Options.
+    /// </summary>
+    public class StartDialogOptions
+    {
+        /// <summary>
+        /// Creates a new instance of the DialogOptions.
+        /// </summary>
+        /// <param name="dialogInputType"></param>
+        /// <param name="botAppId"></param>
+        /// <param name="dialogContext"></param>
+        public StartDialogOptions(DialogInputType dialogInputType, string botAppId, IDictionary<string, object> dialogContext)
+        {
+            DialogInputType = dialogInputType;
+            BotAppId = botAppId;
+            DialogContext = dialogContext;
+            DialogId = Guid.NewGuid().ToString();
+        }
+        /// <summary>
+        /// Creates a new instance of the DialogOptions.
+        /// </summary>
+        /// <param name="dialogInputType"></param>
+        /// <param name="botAppId"></param>
+        /// <param name="dialogContext"></param>
+        /// /// <param name="dialogId"></param>
+        public StartDialogOptions(string dialogId, DialogInputType dialogInputType, string botAppId, IDictionary<string, object> dialogContext)
```

Comment:
if dialogID is optional, but dont want user to update this after consruction, maybe consider making it optional in constructor

Reply:
I believe it was provided as an override so that the options bag would start with the dialogId and not end with it